contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+            "Project ID does not exist"
+        );
+        _;
+    }
+
+    modifier onlyUnlocked(uint256 _projectId) {
+        // Note: calling `_projectUnlocked` enforces that the `_projectId`
+        //       passed in is valid.`
+        require(_projectUnlocked(_projectId), "Only if unlocked");
+        _;
+    }
+
+    modifier onlyAdminACL(bytes4 _selector) {
+        require(
+            adminACLAllowed(msg.sender, address(this), _selector),
+            "Only Admin ACL allowed"
+        );
+        _;
+    }
+
+    modifier onlyArtist(uint256 _projectId) {
+        require(
+            msg.sender == projectIdToFinancials[_projectId].artistAddress,
+            "Only artist"
+        );
+        _;
+    }
+
+    modifier onlyArtistOrAdminACL(uint256 _projectId, bytes4 _selector) {
+        require(
+            msg.sender == projectIdToFinancials[_projectId].artistAddress ||
+                adminACLAllowed(msg.sender, address(this), _selector),
+            "Only artist or Admin ACL allowed"
+        );
+        _;
+    }
+
+    /**
+     * This modifier allows the artist of a project to call a function if the
+     * owner of the contract has renounced ownership. This is to allow the
+     * contract to continue to function if the owner decides to renounce
+     * ownership.
+     */
+    modifier onlyAdminACLOrRenouncedArtist(
+        uint256 _projectId,
+        bytes4 _selector
+    ) {
+        require(
+            adminACLAllowed(msg.sender, address(this), _selector) ||
+                (owner() == address(0) &&
+                    msg.sender ==
+                    projectIdToFinancials[_projectId].artistAddress),
+            "Only Admin ACL allowed, or artist if owner has renounced"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract.
+     * @param _tokenName Name of token.
+     * @param _tokenSymbol Token symbol.
+     * @param _randomizerContract Randomizer contract.
+     * @param _adminACLContract Address of admin access control contract, to be
+     * set as contract owner.
+     * @param _startingProjectId The initial next project ID.
+     * @param _autoApproveArtistSplitProposals Whether or not to always
+     * auto-approve proposed artist split updates.
+     * @dev _startingProjectId should be set to a value much, much less than
+     * max(uint248), but an explicit input type of `uint248` is used as it is
+     * safer to cast up to `uint256` than it is to cast down for the purposes
+     * of setting `_nextProjectId`.
+     */
+    constructor(
+        string memory _tokenName,
+        string memory _tokenSymbol,
+        address _renderProviderAddress,
+        address _platformProviderAddress,
+        address _randomizerContract,
+        address _adminACLContract,
+        uint248 _startingProjectId,
+        bool _autoApproveArtistSplitProposals
+    )
+        ERC721_PackedHashSeed(_tokenName, _tokenSymbol)
+        onlyNonZeroAddress(_randomizerContract)
```

Comment:
since we now specify `_renderProviderAddress` & `_platformProviderAddress` as input addresses, we should add a `onlyNonZeroAddress()` modifier for both of them.

Reply:
Yeah great point, realized we should also have for `_adminACLContract` – updated accordingly (leaving this unresolved to make sure you see the note re: `_adminACLContract`).